src/system_contract.rs

```diff
@@ -75,16 +106,56 @@ impl SystemContract {
                 accounts[0].tokens -= tokens;
                 accounts[1].tokens += tokens;
             }
+            SystemContract::Load { contract_id, name } => {
+                println!("LoadContract");
+                unsafe {
+                    if TEMPORARY.is_none() {
+                        TEMPORARY = Some(Temporary::new());
+                    }
+                    match &TEMPORARY {
+                        Some(temporary) => {
+                            let mut loaded_contracts = temporary.loaded_contracts.write().unwrap();
+                            loaded_contracts.insert(contract_id, DynamicContract::new(name));
+                        }
+                        None => println!("not able to load program {}", name),
+                    }
+                }
+            }
+            SystemContract::Call { contract_id, data } => {
```

Comment:
ah, so instead of `Call` we just dispatch the dynamic contracts from https://github.com/solana-labs/solana/blob/master/src/bank.rs#L363
(which is also why the `Temporary` stuff can just move into `bank.rs`)

Reply:
Something like this?

```rust
        // Call the contract method
        // It's up to the contract to implement its own rules on moving funds
        if SystemContract::check_id(&tx.contract_id) {
            SystemContract::process_transaction(&tx, accounts)
        } else if BudgetContract::check_id(&tx.contract_id) {
            // TODO: the runtime should be checking read/write access to memory
            // we are trusting the hard coded contracts not to clobber or allocate
            BudgetContract::process_transaction(&tx, accounts)
        } else if {
            if let Some(dynamic_contract) = self::check_dynanmic_id(tx.contract_id) {
                dynamic_contract.call(tx, accounts);
            }
        } else {
            return Err(BankError::UnknownContractId(tx.contract_id));
        }
```

Reply:
yep!

Reply:
hmm, yeah.  I think it's ok if the `SystemContract::process_transaction()` is a little special and take in another argument from the bank that allows the system contract to save loaded contracts in.  I wouldn't expose the entire bank to the system contract, just what it needs

Reply:
Like, pass in the hash to be populated?

Reply:
Only slightly less ugly is having system contract call a bank function to register the loaded contract.

Reply:
passing in the hash seems nice